src/test/java/com/puppycrawl/tools/checkstyle/internal/ArchUnitTest.java

```diff
@@ -74,4 +120,36 @@ public void nonProtectedCheckMethodsTest() {
 
         checkMethodsShouldNotBeProtectedRule.check(importedClasses);
     }
+
+    /**
+     * The goal is to ensure all classes in api package are not dependent on classes in util
+     * packages.
+     */
```

Comment:
Can you please add a reason to the comment on why we need such rule?
It's not very obvious.

Reply:
Added.